### PR TITLE
Flatten support for FPGA

### PR DESCRIPTION
**Pull request by manuelburger**
Add support for `Flatten` operator for FPGA expansions by reusing the `Reshape` implementation

### CODE REVIEW

File: daceml/onnx/op_implementations/fpga_implementations.py [hunk 1]
```diff
@@ -1759,23 +1759,52 @@ class FPGAReshape(ONNXForward):
...
+        flatten = False
+
+        # if called from Flatten
+        if "input" in node._in_connectors.keys():
+            input_name = "input"
+            output_name = "output"
+            flatten = True
+
+        if (in_desc_with_name(node, state, sdfg, input_name).dtype !=
+                out_desc_with_name(node, state, sdfg, output_name).dtype):
```
Comment (orausch) [inline daceml/onnx/op_implementations/fpga_implementations.py:1774]: This will already be checked by `node.validate()` before the expansion is called

**Review by orausch — CHANGES_REQUESTED** [review]
Thanks for the quick PR.
Could you please move these two implementations to `op_implementations/pure_implementations.py`? I don't see anything FPGA specific here.

File: daceml/onnx/op_implementations/pure_implementations.py [hunk 2]
```diff
@@ -499,23 +499,58 @@ def prog(X, Y):
...
 class PureReshape(ONNXForward):
+    '''
+        Reshape expansion: this relies on views
+    '''
     @staticmethod
     def forward(node: onnx_op.ONNXOp, state: SDFGState,
                 sdfg: SDFG) -> typing.Union[Node, SDFG]:
-        new_shape = out_desc_with_name(node, state, sdfg, "reshaped").shape
-        node.remove_in_connector("shape")
+        node.validate(sdfg, state)
```
Comment (orausch) [inline daceml/onnx/op_implementations/pure_implementations.py:511]: nit: this is already called before the expansion happens, so it's also not necessary

**Review by orausch — APPROVED** [review]
You might want to format this file with yapf, but otherwise LGTM

**Comment by manuelburger** [comment]
@orausch I have added a test as well now. I could also reopen the PR to merge it directly into master. There is nothing really FPGA specific still in there now. So might as well directly merge into master :man_shrugging:

**Review by TizianoDeMatteis — APPROVED** [review]
Thanks Manuel!

**Comment by orausch** [comment]
> @orausch I have added a test as well now. I could also reopen the PR to merge it directly into master. There is nothing really FPGA specific still in there now. So might as well directly merge into master man_shrugging

Yes, let's do it

**Comment by manuelburger** [comment]
@orausch I think I should make a new PR with these changes. The current one was intended to merge into Tiziano's FPGA branch and contains all of these changes. So I should probably remove those, no?

**Comment by TizianoDeMatteis** [comment]
If you target master branch, then this will bring all the other changes in the fpga branch

**Comment by orausch** [comment]
I started rebasing, but I guess it's easier to make a new PR :smile:

**Comment by manuelburger** [comment]
@orausch @TizianoDeMatteis Already new PR here: https://github.com/spcl/daceml/pull/53

**Pull request closed without merging**